CodeGeneration/Sources/generate-swiftsyntax/templates/swiftparser/ParserEntryFile.swift

```diff
@@ -59,17 +59,34 @@ let parserEntryFile = SourceFileSyntax(leadingTrivia: generateCopyrightHeader(fo
 
   for node in SYNTAX_NODES {
     if let parserFunction = node.parserFunction {
-      DeclSyntax(
-        """
-        extension \(raw: node.name): SyntaxParseable {
-          public static func parse(from parser: inout Parser) -> Self {
-            let node = parser.\(raw: parserFunction)()
-            let raw = RawSyntax(parser.parseRemainder(into: node))
-            return Syntax(raw: raw).cast(Self.self)
+      if node.syntaxKind == "CodeBlockItem" {
+        DeclSyntax(
+          """
+          extension \(raw: node.name): SyntaxParseable {
+            public static func parse(from parser: inout Parser) -> Self {
+              guard let node = parser.\(raw: parserFunction)() else {
+                assertionFailure("Invalid code block item")
```

Comment:
We should not crash when parsing an invalid code block item. Instead, we should form an invalid syntax node. What you could do instead when `parseCodeBlockItem` returns `nil` is to create a `CodeBlockItemSyntax` that contains a `MissingExprSyntax` – or any of the other `Missing*Syntax` nodes, I don’t have a strong preference here.

---

Also, I would prefer not to add a special-case in this code generation loop. Instead, what I think would be the better solution is to have a function `parseNonOptionalCodeBlockItem`, which does what I described above, at the bottom of Parser+Entry.swift and then use that as the `parserFunction`.

Reply:
Awesome, thanks for your suggestions!

I've updated the codes and added corresponding test cases. If there is anything I can do better for this PR, please feel free to let me know.